### PR TITLE
fix(#2445): add c8y proxy config settings

### PR DESCRIPTION
**Pull request by Bravo555**
## TODO

- [x] documentation

## Proposed changes

This PR adds `c8y.proxy.client.host` and `c8y.proxy.client.port` used to connect to running C8y HTTP proxy.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

- #2445

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/references/tedge-cumulocity-proxy.md [hunk 1]
```diff
@@ -11,9 +11,11 @@ thin-edge device.
 It automatically handles authenticating with a JWT, avoiding the need for clients to support MQTT to retrieve this
 information.
 
-The API can be accessed at `http://{ip}:8001/c8y/{c8y-endpoint}`. `ip` is configured using the `c8y.proxy.bind.address`
-configuration
-(the default value is `127.0.0.1`) and the port can be changed by setting `c8y.proxy.bind.port`.
+The Cumulocity HTTP API can be accessed at `http://{host}:{port}/c8y/{c8y-endpoint}`. Configuration settings
+`c8y.proxy.client.host` and `c8y.proxy.client.port` are used to configure `{host}` and `{port}` parts of the base URL
+which will be used by thin-edge components to make requests to the C8y Proxy. `c8y.proxy.bind.address` and
```
Comment (jarhodes314) [inline docs/src/references/tedge-cumulocity-proxy.md:16]: If my understanding is correct, this should stress that the client settings configure the mapper and not the agent (which is what I, as a user, would assume the client setting changes as it is how the equivalent MQTT settings work). I guess the main clue as to this being the case without knowing the implementation details is that the `c8y` settings only affect the mapper as that's the only component that cares about Cumulocity.

Comment (reubenmiller) [reply docs/src/references/tedge-cumulocity-proxy.md:16]: It is defining how clients can reach the endpoint.

The mapper is the only component that uses the setting directly, however the value of the setting will be used within command payloads which are consumed by other clients (e.g. custom device connectors)

Comment (jarhodes314) [reply docs/src/references/tedge-cumulocity-proxy.md:16]: I get that, my point was "there isn't any point setting this on a child device as the value won't be used", which is unlike what happens for the mqtt client settings. It probably doesn't have much of an impact, but the logic behind why it's like that doesn't seem like a thing that's necessarily obvious to a user so might be worth drawing attention to.

Comment (didier-wenzek) [reply docs/src/references/tedge-cumulocity-proxy.md:16]: I agree with @jarhodes314. We must be explicit telling that both `c8y.proxy.bind.address` and `c8y.proxy.client.host` have to be set on the box running the `c8y` mapper. The latter will be forwarded by the c8y mapper to its clients. `c8y.proxy.client.host` can also be directly set on the clients which aim to use the c8y REST API.

Comment (Bravo555) [reply docs/src/references/tedge-cumulocity-proxy.md:16]: My bad, forgot about this ticket. Description tweaked in 0159f91, is it OK now?

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by jarhodes314 — APPROVED** [review]

**Pull request merged**